### PR TITLE
feat: 🎸 refactor app configuration logic

### PR DESCRIPTION
**Pull request by BolajiAyodeji**
### What does this PR do?

This PR changes the access token generation flow and updates the app UX accordingly.

### Description of Task to be completed

- [x] Use Commerce Layer webapp app credentials.
- [x] Use Slack bolt's ExpressReceiver to create a `/callback` endpoint.
- [x] Add a "Auth with Commerce Layer" button logic to the UI.
- [x] Upgrade to the latest js-auth package.
- [x] Add user ability to update the app credentials provided.
- [x] Revert back to integration app credentials.
- [x] Update the app's home tab UI to reflect the token generation changes.
- [x] Add a custom root page UI.
- [x] Add user prompts when the token expires.
- [x] Get organization mode from the decoded access token JWT.

### How should this be manually tested?

Run the app locally and test in the slack app's home tab.

### CODE REVIEW

**Review by malessani — COMMENTED** [review]
I left some comments. Then on Monday I'm going to try it.

File: package-lock.json
[the remark this answers is not in the document]
Comment (BolajiAyodeji) [reply package-lock.json:1]: Sure. I'd have to check first the added significant benefits of using `pnpm` over `npm`. I'm not aware of any yet (you can share also, please). Grazie.

File: src/types/config.ts [hunk 1]
```diff
@@ -2,10 +2,9 @@ import { CommerceLayerClient } from "@commercelayer/sdk";
 
 export interface ConfigOptions {
   cl: CommerceLayerClient;
+  organizationMode: string;
+  organizationSlug: string;
   BASE_ENDPOINT: string;
   CLIENT_ID: string;
```
[the remark this answers is not in the document]
Comment (BolajiAyodeji) [reply src/types/config.ts:8]: Yes. I'm storing the client ID and endpoint so the user will not have to re-enter them when they want to generate a new token after a year. Is that okay?

[the remark this answers is not in the document]
Comment (BolajiAyodeji) [reply src/types/config.ts:8]: It's only needed to pre-fill the configuration form with the last used data. Probably not too important and can be removed if you think we should.

File: src/types/supabase.ts [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+    scope: string;
+    tokenType: string;
+  };
   mode: string;
-  slug: string;
-  salesClientId: string;
-  integrationClientId: string;
-  salesClientSecret: string;
+  endpoint: string;
+  clientId: string;
```
[the remark this answers is not in the document]
Comment (BolajiAyodeji) [reply src/types/supabase.ts:14]: Yes, same reason as the one in `config.ts`. Any decision we make will affect both eventually (both will get updated).

File: src/types/config.ts [hunk 3]
```diff
@@ -2,10 +2,9 @@ import { CommerceLayerClient } from "@commercelayer/sdk";
 
 export interface ConfigOptions {
   cl: CommerceLayerClient;
+  organizationMode: string;
+  organizationSlug: string;
   BASE_ENDPOINT: string;
```
[the remark this answers is not in the document]
Comment (BolajiAyodeji) [reply src/types/config.ts:7]: By notation are you referring to the naming convention of the attribute keys, e.g., `ORGANIZATION_MODE`?

**Comment by BolajiAyodeji** [comment]
> I left some comments. Then on Monday I'm going to try it.

Grazie, Matteo. Do let me know if you need any help setting up the Slack app.

File: package-lock.json
[the remark this answers is not in the document]
Comment (malessani) [reply package-lock.json:1]: Sure, here the context. During our frontend weekly calls we decided to use `pnpm` on all our project

[the remark this answers is not in the document]
Comment (BolajiAyodeji) [reply package-lock.json:1]: Sounds good to me! Thanks for the extra context. I'll try not to take more of your time with a huddle and will proceed to look more into it and migrate 😅.

[the remark this answers is not in the document]
Comment (malessani) [reply package-lock.json:1]: There is an easy way to import packages in pnpm lock file

[the remark this answers is not in the document]
Comment (BolajiAyodeji) [reply package-lock.json:1]: Sounds cool!. I'll migrate in another PR once this is merged.

**Review by malessani — APPROVED** [review]
👍

**Pull request merged**